include/jemalloc/internal/thread_event.h

```diff
@@ -0,0 +1,137 @@
+#ifndef JEMALLOC_INTERNAL_THREAD_EVENT_H
+#define JEMALLOC_INTERNAL_THREAD_EVENT_H
+
+#include "jemalloc/internal/tsd.h"
+
+/*
+ * Maximum threshold on thread_allocated_next_event_fast, so that there is no
+ * need to check overflow in malloc fast path. (The allocation size in malloc
+ * fast path never exceeds SC_LOOKUP_MAXCLASS.)
+ */
+#define THREAD_ALLOCATED_NEXT_EVENT_FAST_MAX				\
+    (UINT64_MAX - SC_LOOKUP_MAXCLASS + 1U)
+
+/*
+ * If we don't set an upper limit for event triggering which is significantly
```

Comment:
That looks great! I only have two minor modifications:

(a) I think the main reason here is not that we want to avoid overflow check on the fast path, but it is that overly long event wait times would make `thread_allocated_next_event` wrap around first. In other words, a zero `thread_allocated_next_event_fast` is not common and is not the main factor driving malloc to medium path.

(b) The max interval also makes sure that we don't keep falling back for too long when there's no active event, based on our new design.